examples/js/tests/unfavorable_prices.ts

```diff
@@ -37,67 +37,112 @@ async function setupAsset() {
   await depositAssets({ USDT: "100.0", ETH: "50.0" }, bidUser);
 }
 
-async function wrongBidPriceTest() {
-  const askOrder = await client.orderPut(askUser, market, ORDER_SIDE_ASK, ORDER_TYPE_LIMIT, /*amount*/ "4", /*price*/ "2", fee, fee);
-  const bidOrder = await client.orderPut(bidUser, market, ORDER_SIDE_BID, ORDER_TYPE_LIMIT, /*amount*/ "4", /*price*/ "10", fee, fee);
-
-  console.log("ask order id", askOrder.id);
-  console.log("bid order id", bidOrder.id);
-
-  // the trade should have happend with a price of "2"
+async function checkAskUserBalance() {
   const balanceAsk = await client.balanceQuery(askUser);
-  let usdtBalance = balanceAsk.get("USDT");
-  let ethBalance = balanceAsk.get("ETH");
+  const usdtBalance = balanceAsk.get("USDT");
+  const ethBalance = balanceAsk.get("ETH");
   assertDecimalEqual(usdtBalance.available, "108");
   assertDecimalEqual(usdtBalance.frozen, "0");
   assertDecimalEqual(ethBalance.available, "46");
   assertDecimalEqual(ethBalance.frozen, "0");
+}
 
+async function checkBidUserBalance() {
   const balanceBid = await client.balanceQuery(bidUser);
-  usdtBalance = balanceBid.get("USDT");
-  ethBalance = balanceBid.get("ETH");
+  const usdtBalance = balanceBid.get("USDT");
+  const ethBalance = balanceBid.get("ETH");
   assertDecimalEqual(usdtBalance.available, "92");
   assertDecimalEqual(usdtBalance.frozen, "0");
   assertDecimalEqual(ethBalance.available, "54");
   assertDecimalEqual(ethBalance.frozen, "0");
+}
+
+async function wrongBidPriceTest() {
+  const askOrder = await client.orderPut(askUser, market, ORDER_SIDE_ASK, ORDER_TYPE_LIMIT, /*amount*/ "4", /*price*/ "2", fee, fee);
+  const bidOrder = await client.orderPut(bidUser, market, ORDER_SIDE_BID, ORDER_TYPE_LIMIT, /*amount*/ "4", /*price*/ "10", fee, fee);
+
+  console.log("ask order id", askOrder.id);
+  console.log("bid order id", bidOrder.id);
+
+  await checkAskUserBalance();
+  await checkBidUserBalance();
 
   console.log("wrongBidPriceTest successfull");
 }
 
+async function wrongBidPriceWithBigVolumeTest() {
+  const askOrder = await client.orderPut(askUser, market, ORDER_SIDE_ASK, ORDER_TYPE_LIMIT, /*amount*/ "4", /*price*/ "2", fee, fee);
+  const bidOrder = await client.orderPut(bidUser, market, ORDER_SIDE_BID, ORDER_TYPE_LIMIT, /*amount*/ "10", /*price*/ "10", fee, fee);
+
+  console.log("ask order id", askOrder.id);
+  console.log("bid order id", bidOrder.id);
+
+  await checkAskUserBalance();
+  const balanceBid = await client.balanceQuery(bidUser);
+  const usdtBalance = balanceBid.get("USDT");
+  const ethBalance = balanceBid.get("ETH");
+  assertDecimalEqual(usdtBalance.available, "32");
+  assertDecimalEqual(usdtBalance.frozen, "60");
+  assertDecimalEqual(ethBalance.available, "54");
+  assertDecimalEqual(ethBalance.frozen, "0");
+
+  const openOrders = await client.orderQuery(bidUser, market);
+
+  assertDecimalEqual(openOrders.orders[0].remain, "6");
```

Comment:
Could you also assert that the price of the remaining bid order is 10 ?

Because I'm not a 100% sure that this will be the case, but it's what I'm assuming.